jlexer/lexer.go

```diff
@@ -532,34 +531,6 @@ func (r *Lexer) Raw() []byte {
 	return r.Data[r.start:r.pos]
 }
 
-// IsStart returns whether the lexer is positioned at the start
-// of an input string.
-func (r *Lexer) IsStart() bool {
-	return r.pos == 0
-}
-
-// Consumed reads all remaining bytes from the input, publishing an error if
-// there is anything but whitespace remaining.
-func (r *Lexer) Consumed() {
-	if r.pos > len(r.Data) {
-		return
-	}
-
-	for _, c := range r.Data[r.pos:] {
-		if c != ' ' && c != '\t' && c != '\r' && c != '\n' {
-			r.err = &LexerError{
```

Comment:
^ this here is the problem, Re-assigning `r.err` even if `r.err != nil`.

This error message of "invalid character" is really confusing because there is no invalid character at all.

I have an error message about "field foo being required" but somewhere the code did not short circuit properly and started complaining about invalid characters instead.